From 2c0f5d4121f6048cc3686cdf0e0ed80de9087c8e Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 21 May 2023 01:26:03 +0200
Subject: [PATCH 219/469] usb: typec: anx7688: Don't use I2C regulator

anx7688 never disables any regulators across suspend. So we don't want
to keep this one enabled, beceuase on Pinephone it's also powering
touch controller which is hard to disable during sleep, so it needs
its power removed.

(Alternative here would be to disable I2C bus regulator in this driver
during suspend, since it's not needed for I2C comm during suspend. But
just not using it is easier for now.)

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/usb/typec/anx7688.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/typec/anx7688.c b/drivers/usb/typec/anx7688.c
index 3fa1eb7edd7b..0465d7808673 100644
--- a/drivers/usb/typec/anx7688.c
+++ b/drivers/usb/typec/anx7688.c
@@ -160,7 +160,9 @@ static const char * const anx7688_supply_names[] = {
         "dvdd18",
         "avdd10",
         "dvdd10",
-        "i2c",
+	//XXX: this would block I2C power supply from being disabled during suspend
+	//by other drivers, so let's not enable I2C power in this driver.
+	//"i2c",
         "hdmi_vt",
 
         "vconn", // power for VCONN1/VCONN2 switches
-- 
2.34.1

